### PR TITLE
test(js): Throw on `console.error` during js tests

### PR DESCRIPTION
**Pull request by billyvg**
jest will now fail whenever `console.error` is called - there is a list of ignored messages.

### CODE REVIEW

**Comment by billyvg** [comment]
Please check out the example error logs that get displayed and lmk your thoughts @getsentry/app-frontend 

https://travis-ci.org/github/getsentry/sentry/jobs/667347197?utm_medium=notification&utm_source=github_status

**Comment by billyvg** [comment]
We should make sure getsentry doesn't die

**Comment by markstory** [comment]
Looks like quite a few tests will need to be fixed before we can get this merge.d

**Comment by billyvg** [comment]
@markstory oops yeah, the errors there are intentional to show what the logs would look like

**Comment by billyvg** [comment]
Logs are more readable now, see https://travis-ci.org/github/getsentry/sentry/jobs/667858645?utm_medium=notification&utm_source=github_status

**Comment by billyvg** [comment]
Note this won't affect `getsentry` because it doesn't share jest config

File: tests/js/throw-on-react-error.js [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
   if (
     typeof message === 'string' &&
     message.indexOf(REPEATED_ERROR) !== 0 &&
-    BAD_ERRORS_REGEX.test(message)
+    !isIgnored
   ) {
     originalConsoleError(message, ...args);
-    throw new Error(message);
-  } else if (!BAD_ERRORS_REGEX.test(message)) {
-    originalConsoleError(message, ...args);
```
Comment (markstory) [inline tests/js/throw-on-react-error.js]: :clap: This should cut down on the noisy duplicates.

**Review by markstory — APPROVED** [review]

**Pull request merged**